src/main/riddl/Tenant/tenant.riddl

```diff
@@ -134,22 +36,31 @@ context TenantContext is {
                 meta: TenantContext.MetaInfo
             }
             handler DraftTenantHandler is {
-                on command ActivateTenant {
+                on command TenantContext.EditInfo {
+                    then "Copy fields from TenantContext.EditInfo into Draft info fields"
+                    and "adjust meta fields to reflect the change set"
+                    and "yield event TenantContext.InfoUpdated"
+                }
+                on command TenantContext.UpdateStatus {
                     example Success {
-                        then "Morph Tenant to Active state"
+                        then "Morph Tenant to desired state"
                         and "Set Tenant meta data to reflect the change"
-                        and "yield TenantActivated event"
+                        and "yield TenantContext.StatusUpdated event"
+                    }
+                    example SameStatus {
+                        when "Tenant already is in desired state"
+                        then error "Tenant is already in desired state"
                     }
                     example NoBillingAccount {
-                        when "BillingAccount is null"
+                        when "Desired status is Active and BillingAccount is null"
```

Comment:
As I read this the thought that pops to mind is it seems you are trying to make a generic UpdateStatus command, when in reality we are only dealing with the Draft state. I think we can be much narrower in our handler definitions given this observation.

Reply:
I included these error cases because I thought BillingAccount, DistributionAccount were required fields for getting out of Draft state. Please correct me if I'm wrong :)

Reply:
BillingAccount and DistributionAccount are out of scope for the Kalix-Study. We will need them to be inserted before release. I think we are discussing two different ideas here. Perhaps a call to discuss would be in order to quickly understand both view points.